code/go/0chain.net/smartcontract/minersc/dkg.go

```diff
@@ -224,8 +224,13 @@ func GetPhaseNode(statectx cstate.StateContextI) (
 
 func (msc *MinerSmartContract) setPhaseNode(balances cstate.StateContextI,
 	pn *PhaseNode, gn *GlobalNode, t *transaction.Transaction) error {
+
+	isViewChange, err := config.Configuration().ChainConfig.ReadValue("ViewChange")
```

Comment:
Add `Config()` method to the `cstate.StateContextI` to get config from chain struct, or add the config access function to the `MinerSmartContract` struct. For example, 

```go
type GetConfigFunc func() *config.Config

type MinerSmartContract struct {
    getConfig smartcontract.GetConfigFunc
}

Init the MinerSmartContract with the function to get config from Chain when the smart contract is called. Then in the `setPhaseNode` method, we can run `msc.getConfig()` to get the config directly. 

```


Do this to all other smart contracts that need to access the config.

Reply:
Yes, the dependence circle issue could happen as the `Config` interface is defined in `chain` package, let's move it and the `ConfigImpl` all to `chaincore/config` package. Then have `chain` struct to embed the `config.Config` interace. One thing to be noted is the `Update()` method: https://github.com/0chain/0chain/blob/89884cb0fdf825d04df1c1d6967b633512c8baf4/code/go/0chain.net/chaincore/chain/config.go#L86
It uses the `minersc.GlobalSettings`, which don't have to be in `minersc` package, so we can move it to `chain/config` package too. In this way, both `chain` and smart contracts can access the config directly, and there won't be too much of changes.